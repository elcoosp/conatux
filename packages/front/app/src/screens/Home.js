import React from 'react'
import { Screen, Links } from '@:/screens'
import { ROUTES } from '@:/constants'

const Home = () => (
  <Screen title={ROUTES.Home.label}>
    <Links.Auth.SignUp />
  </Screen>
)

export default Home
