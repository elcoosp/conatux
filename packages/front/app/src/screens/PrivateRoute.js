import React from 'react'
import P from 'prop-types'
import { ROUTES } from '@:/constants'
import { Route, Redirect } from 'react-router-dom'

const isAuth = () => false
const PrivateRoute = ({ component: Component, redirectTo, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      isAuth() ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: redirectTo || ROUTES.Auth.Login.path,
            state: { from: props.location }
          }}
        />
      )
    }
  />
)

PrivateRoute.propTypes = {
  redirectTo: P.string
}

export default PrivateRoute
