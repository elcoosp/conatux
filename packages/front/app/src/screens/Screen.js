import React from 'react'
import P from 'prop-types'

const Screen = ({ title, children }) => (
  <>
    <h1>{title}</h1>
    {children}
  </>
)

Screen.propTypes = { title: P.string.isRequired, children: P.node.isRequired }

export default Screen
