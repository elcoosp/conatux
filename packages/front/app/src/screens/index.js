export { default as Home } from './Home'
export { default as Links } from './Links'
export { default as Screen } from './Screen'
export { default as Profile } from './me/Profile'
export { default as SignUp } from './auth/SignUp'
