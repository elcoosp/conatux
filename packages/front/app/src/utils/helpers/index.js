export const pipe = {
  execAsync: (...funcs) => context =>
    funcs.reduce(async (ctx, func) => {
      await func(ctx)
      return ctx
    }, context)
}
