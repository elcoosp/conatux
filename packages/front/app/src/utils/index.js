export { default as testing } from './testing'
export { default as linkify } from './linkify'
