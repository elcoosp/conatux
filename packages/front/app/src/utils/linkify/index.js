import React from 'react'
import { Link } from 'react-router-dom'

export const labelize = (path, label) => ({
  path,
  label
})

export const fromRoutes = (routesTree, componentName) => {
  if (['label', 'path'].every(prop => routesTree.hasOwnProperty(prop))) {
    const { path, label } = routesTree
    const link = props => (
      <Link to={path} {...props}>
        {label}
      </Link>
    )
    link.displayName = componentName
    return link
  }

  return Object.entries(routesTree).reduce(
    (links, [routeName, nestedRoute]) => {
      const componentName =
        routeName.slice(0, 1).toUpperCase() + routeName.slice(1)
      links[componentName] = fromRoutes(nestedRoute, componentName)
      return links
    },
    {}
  )
}
