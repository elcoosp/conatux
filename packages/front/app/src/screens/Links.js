import { ROUTES } from '@:/constants'
import { fromRoutes } from '@:/utils/linkify'

export default fromRoutes(ROUTES)
