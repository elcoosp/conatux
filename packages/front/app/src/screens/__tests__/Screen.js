import { desc } from '@:/utils/testing'
import { Screen } from '@:/screens'

desc.compWithSnap({ Screen }, { title: 'Title', children: 'Content' }, () => {})
