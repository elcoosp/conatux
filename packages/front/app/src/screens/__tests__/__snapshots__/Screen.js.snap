// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> should match snapshot 1`] = `
<Fragment>
  <h1>
    Title
  </h1>
  Content
</Fragment>
`;
